### PR TITLE
add DeprecatedRpcTest

### PR DESCRIPTION
**Pull request by mess110**
Hi John,

This PR adds a functional test for https://github.com/bitcoin/bitcoin/pull/11031 .

Please advise if something is missing or wrong.

Thanks,
mess110

### CODE REVIEW

File: test/functional/deprecated_rpc.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test deprecation of RPC calls."""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
```
Comment (jnewbery) [inline test/functional/deprecated_rpc.py]: nit: prefer explicit imports over wildcard imports. Here, you're just using `assert_raises_jsonrpc` from util.py, so you can call:

```python
from test_framework.util import assert_raises_jsonrpc
```

File: test/functional/deprecated_rpc.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class DeprecatedRpcTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 2
+        self.setup_clean_chain = True
+
+    def setup_network(self):
```
Comment (jnewbery) [inline test/functional/deprecated_rpc.py]: You don't need to override this method if you set `self.extra_args` in the `__init__()` method. Add the following line to the end of `_init__()` and remove the `setup_network()` method override:

```python
self.extra_args = [[], ["-enablerpcmethod=estimatefee"]]
```

File: test/functional/deprecated_rpc.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        self.nodes = []
+        self.nodes.append(self.start_node(0, self.options.tmpdir))
+        self.nodes.append(self.start_node(1, self.options.tmpdir,
+            ["-enablerpcmethod=estimatefee"]))
+
+    def run_test (self):
```
Comment (jnewbery) [inline test/functional/deprecated_rpc.py]: nit: please no space after the method name

File: test/functional/deprecated_rpc.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        self.log.info("estimatefee: Shows deprecated message")
+        assert_raises_jsonrpc(-32, 'estimatefee is deprecated',
```
Comment (jnewbery) [inline test/functional/deprecated_rpc.py]: no need for a lambda here, `assert_raises_jsonrpc()` should be called like this:

```python
assert_raises_jsonrpc(-32, 'estimatefee is deprecated', self.nodes[0].estimatefee, 1)
```

**Review by jnewbery — COMMENTED** [review]
Great, thanks for this @mess110.

Looks good - I've added a couple of nits. If you make those changes and push, then I'll accept this commit into my PR.

Thanks for adding this test!

**Comment by mess110** [comment]
@jnewbery thanks for the review

**Pull request merged**